### PR TITLE
CFL Development Pull Request

### PR DESCRIPTION
**Pull request by TalbenXu**
Support Attribute Grammar
Support EBNF Transformation
Support Grammar Print out
Decoupling with additional builder

### CODE REVIEW

File: include/CFL/CFLAlias.h [hunk 1]
```diff
@@ -57,39 +59,37 @@ class CFLAlias : public PointerAnalysis
...
+            Map<std::string, SVF::CFLGraph::Symbol> ConstMap =  {{"Addr",0}, {"Copy", 1},{"Store", 2},{"Load", 3},{"Gep_i", 4},{"Vgep", 5},{"Addrbar",6}, {"Copybar", 7},{"Storebar", 8},{"Loadbar", 9},{"Gepbar_i", 10},{"Vgepbar", 11}};
+            GrammarBase *generalGrammar = gReader.build(ConstMap);
             ConstraintGraph *consCG = new ConstraintGraph(svfir);
-            svfir->dump("SVFIR");
-            // Can be put in build, copy from general graph
-            graph->label2SymMap = ConstMap;
-            graph->buildBigraph(consCG);
-            graph->dump("PAG");
+            graph = cflGraphBuilder.buildBigraph(consCG);
+            cflChecker.check(generalGrammar, graph);
```
Comment (yuleisui) [inline include/CFL/CFLAlias.h:74]: The CFL checker should be able to check any grammar with a graph including the grammar after normalization.

File: include/CFL/CFLChecker.h [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+ *  Created on: May 23, 2022
+ *      Author: Pei Xu
+ */
+
+#include "CFL/CFLGrammar.h"
+#include "Graphs/CFLGraph.h"
+namespace SVF
+{
+
+class CFLChecker
```
Comment (yuleisui) [inline include/CFL/CFLChecker.h:35]: Should rename this to be `CFLGramGraphChecker`.

File: include/CFL/CFLAlias.h (hunk 1 above)
Comment (TalbenXu) [reply include/CFL/CFLAlias.h:74]: GramGraphChecker is currently receiving the GrammarBase should be also able received other child class like cflgrammar.

File: include/CFL/CFLGrammar.h [hunk 3]
```diff
@@ -93,26 +145,27 @@ class CFLGrammar : public GrammarBase
         return it!=secondRHS2Prods.end();
     }
 
-    const Productions& getProdsFromSingleRHS(Symbol sym) const
-    {
+    Productions getProdsFromSingleRHS(Symbol sym) const
```
Comment (yuleisui) [inline include/CFL/CFLGrammar.h:148]: Why remove const? Using `const` is always a good practice.

Comment (yuleisui) [reply include/CFL/CFLGrammar.h:148]: Try to add `const` wherever possible to make sure code is always readable

File: include/CFL/CFLAlias.h (hunk 1 above)
Comment (yuleisui) [reply include/CFL/CFLAlias.h:74]: Any way to check whether the normalized grammar is correct and consistent with the graph?

**Pull request merged**